model/project_aliases.go

```diff
@@ -71,38 +71,110 @@ type ProjectAlias struct {
 type ProjectAliases []ProjectAlias
 
 // FindAliasesForProject fetches all aliases for a given project
-func FindAliasesForProject(projectID string) ([]ProjectAlias, error) {
-	out := []ProjectAlias{}
+func FindAliasesForProject(projectID string, version string) ([]ProjectAlias, error) {
+	var out []ProjectAlias
 	q := db.Query(bson.M{
 		projectIDKey: projectID,
 	})
 	err := db.FindAllQ(ProjectAliasCollection, q, &out)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding project aliases")
+	}
+	parserProject, err := ParserProjectByVersion(projectID, version)
+	if err != nil {
+		return []ProjectAlias{}, errors.Wrap(err, "error finding project parser")
+	}
+	if parserProject != nil && len(parserProject.Aliases) > 0 {
+		parserProjectAliases := aliasesToMap(parserProject.Aliases)
+		dbAliases := aliasesToMap(out)
+		return mergeAliasMaps(parserProjectAliases, dbAliases), nil
```

Comment:
I like this implementation! Can mergeAliasMaps instead take in just the lists and be "mergeAliases" and create the maps inside, to move more of the implementation out?